File: BACKEND/BE/Models/Evento.cs
using Azure;
using BE.Models;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace BE.Models
{
    public class Evento
    {
        public int ID { get; set; }

        public string Nombre { get; set; }

        [DataType(DataType.Date)]
        [DisplayFormat(DataFormatString = "{0:dd-MM-yyyy}", ApplyFormatInEditMode = true)]
        public DateTime Fecha { get; set; }

        public string Lugar { get; set; } 

        public string Estado { get; set; } 

        public byte[]? Imagen { get; set; }


        public ICollection<Categoria> Categorias { get; set; }


        public virtual ICollection<EventoDistancia> EventoDistancias { get; set; }



        public virtual ICollection<Comentario> Comentarios { get; set; } 


        public virtual ICollection<Inscripcion> Inscripciones { get; set; } 

        public virtual TipoEvento Tipo { get; set; }
        [ForeignKey("Tipo")]
        public int TipoID { get; set; }
    }
}

File: FRONTEND/FE-CRUDEventos/src/app/interfaces/evento.ts
export interface Evento {
  id: number;
  nombre: string;
  fecha: Date;
  lugar: string;
  hora: string;
  estado: string;
  imagen?: string;
  tipo: Tipo;
}


// Definir el tipo de eventoData
export interface EventoData {
  nombre: string;
  fecha: Date | null;
  lugar: string;
  hora: string;
  eventoDistancias: EventoDistancia[]; // Puedes definir un tipo específico si lo necesitas
  tipoID: number | null;
}

// Inicializar eventoData con los valores correctos
let eventoData: EventoData = {
  nombre: '',
  fecha: null,
  lugar: '',
  hora: '',
  eventoDistancias: [],
  tipoID: null
};

// Definición de la interfaz para la respuesta completa
export interface EventoResponse {
  evento: Evento;
  categorias: Categoria[];
  distancias: Distancia[];
}

export interface Tipo {
  id: number;
  descripcion: string;
}

export interface Categoria {
  id: number;
  edadInicio: number;
  edadFin: number;
}

export interface Distancia {
  id: number;
  distanciaID: number;
  km: number;
  precio: number;
}
export interface EventoDistancia {
  distanciaID: number;
  precio: number;
}
